src/cmd/singularity/cli/build.go

```diff
@@ -102,6 +102,8 @@ var BuildCmd = &cobra.Command{
 					authWarning != auth.WarningTokenTooShort {
 					if authToken != "" {
 						b = build.NewRemoteBuilder(args[0], def, false, remoteURL, authToken)
+					} else {
+						return
```

Comment:
I'm really not sure about the logic in this section. On the one hand, if `authWarning` is set to `auth.WarningEmptyToken`, it `Fatalf`s. But if the token is `""` (which is empty?) it returns? 

Shouldn't this just `Fatalf` if `authToken` is not valid for any reason? Why the divergent strategies? How can `authToken == ""` and `authWarning != auth.WarningEmptyToken` (as seems to be the case right now?)

Reply:
`auth.Warning` is set when reading the tokenFile, but we also leave room for reading the token from an env var. that's why the double check